### PR TITLE
fix: improve image type detection

### LINKED ISSUE #16668: Cannot create a new community -  message:unsupported file type when trying to create a community selecting certain images
# Bug Report

## Description

When the user selects an unsupported image on the community logo or banner, an error message shows on the app's console but no indication of error shows on the UI. This is confusing to the user because they do not know why they are unable to create a community.

## Steps to reproduce

1. Initiate the flow to create a new community
2. Choose an image such as an image attached
3. Fill in the required fields to create a community, and add the attached file for the Community Logo and Community banner.

[ff.jpeg.zip](https://github.com/user-attachments/files/17577642/ff.jpeg.zip)

#### Expected behavior
If the selected image is incompatible, an error message to show on the app UI indicates this is not compatible.
Minimum image requirements (resolution, size, etc) should be provided to the user

#### Actual behavior
The image is not supported, and no error message or warning is shown to the user.
The pop-up to create a community closes but the community is not created
Error messages show on the app console.

Image

https://github.com/user-attachments/assets/42b4845a-ff6f-4db2-98ec-062ca18cd1bf

[app_20241030_151356.log](https://github.com/user-attachments/files/17577627/app_20241030_151356.log)

```
DBG 2024-10-30 18:49:48.862Z NewBE_callPrivateRPC                       topics="rpc" tid=80050700 file=core.nim:27 rpc_method=wakuext_curatedCommunities
DBG 2024-10-30 18:49:48.870Z [threadpool task thread] initiating task   topics="task-threadpool" tid=80050701 file=threadpool.nim:56 messageType=AsyncAddCentralizedMetricIfEnabledTaskArg:ObjectType threadid=80050701 task="{\"$type\":\"AsyncAddCentralizedMetricIfEnabledTaskArg:ObjectType\",\"eventName\":\"navigation\",\"eventValueJson\":\"{\\\"viewId\\\":\\\"communitiesPortal\\\"}\",\"vptr\":105553145123328,\"slot\":\"onCentralizedMetricAddedIdEnabled\"}"
WRN 2024-10-30 18:49:53.505Z qt warning                                 topics="qt" tid=80050686 text="Unable to assign CreateCommunityPopup_QMLTYPE_2183 to QQuickItem" file=qrc:/app/AppLayouts/Communities/popups/CreateCommunityPopup.qml:377 category=default
WRN 2024-10-30 18:50:49.005Z qt warning                                 topics="qt" tid=80050686 text="QML StatusEmoji: Cannot open: qrc:/assets/twemoji/svg/39e.svg" file=qrc:/StatusQ/Controls/StatusCommunityTag.qml:36 category=default
DBG 2024-10-30 18:51:00.039Z NewBE_callPrivateRPC                       topics="rpc" tid=80050686 file=core.nim:27 rpc_method=wakuext_createCommunity
ERR 2024-10-30 18:51:00.041Z rpc response error                         topics="rpc" tid=80050686 file=core.nim:36 err="\nstatus-go error [methodName:wakuext_createCommunity, code:-32000, message:unsupported file type ]\n"
ERR 2024-10-30 18:51:00.041Z error doing rpc request                    topics="rpc" tid=80050686 file=core.nim:39 methodName=wakuext_createCommunity exception="\nstatus-go error [methodName:wakuext_createCommunity, code:-32000, message:unsupported file type ]\n"
ERR 2024-10-30 18:51:00.041Z Error creating community                   topics="community-service" tid=80050686 file=service.nim:1225 msg="\nstatus-go error [methodName:wakuext_createCommunity, code:-32000, message:unsupported file type ]\n"
```

### Additional Information

- Status desktop version: 2.30 and 2.31 rc-2
- Operating System: MacOS Sonoma

### PR DESCRIPTION
**Pull request by caybro**
### What does the PR do

- use the same approach as status-go to detect the image type, relying on "magic" type matching instead of looking at the file extension (now using C++ and QMime*)
- add a little error popup when the user tries to upload an unsupported image type while creating/editing a community
- expose all the image related properties from the C++ backend instead of constructing and duplicating them in QML
- cleanup some unused/dead code

Fixes #16668

### Affected areas

Community settings, StatusChatInput, Image DND

### Architecture compliance

- [x] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Trying to change the community logo, using an AVIF (unsupported) image:

https://github.com/user-attachments/assets/8978a02a-fce5-4ca0-8a32-ea7ba9424373

Trying to upload an unsupported image to chat:
image

Listing all supported image types (including uppercase extension):
image

### CODE REVIEW

**Comment by caybro** [comment]
@virginiabalducci pls give it a try :)

**Review by jrainville — APPROVED** [review]
Nice job!

File: ui/imports/shared/popups/ImageCropWorkflow.qml [hunk 1]
```diff
@@ -39,14 +42,37 @@ Item {
...
-        nameFilters: [qsTr("Supported image formats (%1)").arg(Constants.acceptedDragNDropImageExtensions.map(img => "*" + img).join(" "))]
+        nameFilters: [qsTr("Supported image formats (%1)").arg(UrlUtils.validImageNameFilters)]
         onAccepted: {
             if (fileDialog.fileUrls.length > 0) {
-                cropImage(fileDialog.fileUrls[0])
+                const url = fileDialog.fileUrls[0]
+                if (Utils.isValidDragNDropImage(url))
+                    cropImage(url)
+                else
+                    errorDialog.open()
```
[the remark this answers is not in the document]
Comment (caybro) [reply ui/imports/shared/popups/ImageCropWorkflow.qml:52]: Yeah, still cannot grasp how this is valid code, esp. after switching from C++ 😁 I'd rather not though; this will work until someone adds more code below, after which it's easy to miss the early return

[the remark this answers is not in the document]
Comment (jrainville) [reply ui/imports/shared/popups/ImageCropWorkflow.qml:52]: It's pretty common to early return in JS, but as you wish

**Comment by virginiabalducci** [comment]
Thanks for taking a look at this @caybro 😄 
What about these images? Is this a different issue? about the wrong size? 

The image does not load, and though there is no error message to the user it's clear that the image selected cannot be used, but would be nice to have some message for the user to choose a different image. If this is a different issue I can open a new bug.

[images.zip](https://github.com/user-attachments/files/17600729/images.zip)

```
ERR 2024-11-01 14:43:36.635Z qt error                                   topics="qt" tid=82148493 text="Wrong source size. Ensure source is set" file=qrc:/StatusQ/Controls/StatusImageCrop.qml:174 category=qml
WRN 2024-11-01 14:43:42.529Z qt warning                                 topics="qt" tid=82148493 text="QML Image: Error decoding: file:///Users/virginiabalducci/Desktop/images/GettyImages-164855910-1-e1694532950609.jpeg: Unsupported image format" file=qrc:/StatusQ/Controls/StatusImageCrop.qml:348 category=default
ERR 2024-11-01 14:43:47.709Z qt error                                   topics="qt" tid=82148493 text="Wrong source size. Ensure source is set" file=qrc:/StatusQ/Controls/StatusImageCrop.qml:174 category=qml
ERR 2024-11-01 14:43:47.710Z qt error                                   topics="qt" tid=82148493 text="Wrong source size. Ensure source is set" file=qrc:/StatusQ/Controls/StatusImageCrop.qml:174 category=qml
WRN 2024-11-01 14:43:56.327Z qt warning                                 topics="qt" tid=82148493 text="QML Image: Error decoding: file:///Users/virginiabalducci/Desktop/images/National_Black_Cat_Day.jpeg: Unsupported image format" file=qrc:/StatusQ/Controls/StatusImageCrop.qml:348 category=default
ERR 2024-11-01 14:44:01.396Z qt error                                   topics="qt" tid=82148493 text="Wrong source size. Ensure source is set" file=qrc:/StatusQ/Controls/StatusImageCrop.qml:174 category=qml
ERR 2024-11-01 14:44:01.396Z qt error                                   topics="qt" tid=82148493 text="Wrong source size. Ensure source is set" file=qrc:/StatusQ/Controls/StatusImageCrop.qml:174 category=qml
WRN 2024-11-01 14:44:05.057Z qt warning                                 topics="qt" tid=82148493 text="QML Image: Error decoding: file:///Users/virginiabalducci/Desktop/images/final-fantasy-pixel-remaster-art.jpeg: Unsupported image format" file=qrc:/StatusQ/Controls/StatusImageCrop.qml:348 category=default
DBG 2024-11-01 14:46:35.492Z setCommunityInfo                           tid=82148493 file=status_community_link_preview.nim:155 communityId=0x027801b8aaba3f77d3d1d85d9f76b725500d5c39b30f50d77ec12de8f27358e5ba communityName="Grey Cats"
DBG 2024-11-01 14:46:35.492Z setCommunityInfo                           tid=82148493 file=status_community_link_preview.nim:155 communityId=0x027801b8aaba3f77d3d1d85d9f76b725500d5c39b30f50d77ec12de8f27358e5ba communityName="Grey Cats"
WRN 2024-11-01 14:48:40.526Z qt warning                                 topics="qt" tid=82148493 text="QML Image: Error decoding: file:///Users/virginiabalducci/Desktop/images/GettyImages-164855910-1-e1694532950609.jpeg: Unsupported image format" file=qrc:/StatusQ/Controls/StatusImageCrop.qml:348 category=default
```


https://github.com/user-attachments/assets/4fa6c721-149a-4a6f-8b4b-dd18963075da



Otherwise, 'ff.jpeg' that was modified from webp to jpeg correctly shows the error message when trying to use.

I tested with real JPEG images from here https://filesamples.com/formats/jpeg and they all can be selected as community profile or banner picture.

**Comment by caybro** [comment]
> Thanks for taking a look at this @caybro 😄 What about these images? Is this a different issue? about the wrong size?
> 
> The image does not load, and though there is no error message to the user it's clear that the image selected cannot be used, but would be nice to have some message for the user to choose a different image. If this is a different issue I can open a new bug.

Dunno yet, will try with the attached ZIPped images. But the warning clearly says that this image type (whatever it is, but I'm assuming you again renamed them :laughing:) is not supported.

**Comment by caybro** [comment]
```
❯ file GettyImages-164855910-1-e1694532950609.jpeg 
GettyImages-164855910-1-e1694532950609.jpeg: RIFF (little-endian) data, Web/P image, VP8 encoding, 1024x647, Scaling: [none]x[none], YUV color, decoders should clamp

❯ file National_Black_Cat_Day.jpeg 
National_Black_Cat_Day.jpeg: RIFF (little-endian) data, Web/P image

❯ file final-fantasy-pixel-remaster-art.jpeg 
final-fantasy-pixel-remaster-art.jpeg: RIFF (little-endian) data, Web/P image, VP8 encoding, 687x386, Scaling: [none]x[none], YUV color, decoders should clamp
```

**Comment by caybro** [comment]
I mean, even if they have a "jpeg" extension, they are in fact "image/webp" which is supported by Qt and by status-go as well

**Comment by caybro** [comment]
> ```
> ❯ file GettyImages-164855910-1-e1694532950609.jpeg 
> GettyImages-164855910-1-e1694532950609.jpeg: RIFF (little-endian) data, Web/P image, VP8 encoding, 1024x647, Scaling: [none]x[none], YUV color, decoders should clamp
> 
> ❯ file National_Black_Cat_Day.jpeg 
> National_Black_Cat_Day.jpeg: RIFF (little-endian) data, Web/P image
> 
> ❯ file final-fantasy-pixel-remaster-art.jpeg 
> final-fantasy-pixel-remaster-art.jpeg: RIFF (little-endian) data, Web/P image, VP8 encoding, 687x386, Scaling: [none]x[none], YUV color, decoders should clamp
> ```

those "fake" jpegs are "webp" in fact ^^

**Comment by caybro** [comment]
TLDR, made the "webp" support optional based on whether the "libwebp" is available for Qt; on Linux this works our of the box but it seems that on macOS, our packaging is broken and we don't bundle/build the Qt correctly

**Comment by virginiabalducci** [comment]
The fix also works for the scenario where a user tries to upload an unsupported webp image

Screenshot 2024-11-01 at 2 02 41 PM

https://github.com/user-attachments/assets/b7cd65a6-7d7d-46b8-af42-aec6f7b842a2

**Review by virginiabalducci — APPROVED** [review]
